statefun-flink/statefun-flink-core/src/test/java/org/apache/flink/statefun/flink/core/jsonmodule/RemoteModuleTest.java

```diff
@@ -63,13 +59,90 @@ public void testComponents() {
     StatefulFunctionModule module = fromPath(modulePath);
 
     StatefulFunctionsUniverse universe = emptyUniverse();
-    setupUniverse(universe, module, new TestComponentBindersModule());
+    setupUniverse(universe, module, new TestComponentBindersModule(), new HashMap<>());
 
     assertThat(universe.functions(), hasKey(TestComponentBinder1.TEST_FUNCTION_TYPE));
     assertThat(universe.ingress(), hasKey(TestComponentBinder2.TEST_INGRESS.id()));
     assertThat(universe.egress(), hasKey(TestComponentBinder3.TEST_EGRESS.id()));
   }
 
+  @Test
+  public void configuringComponentsShouldResolvePlaceholders() {
+    final AtomicInteger counter = new AtomicInteger();
+    final Map<String, String> configuration = new HashMap<>();
+    configuration.put(TEST_CONFIG_KEY_1, TEST_CONFIG_VALUE_1);
+    configuration.put(TEST_CONFIG_KEY_2, TEST_CONFIG_VALUE_2);
```

Comment:
[Nitpick] From Java 9, there is a factory method `Map.of(...)`:
```
final Map<String, String> configuration = Map.of(
  TEST_CONFIG_KEY_1, TEST_CONFIG_VALUE_1,
  TEST_CONFIG_KEY_2, TEST_CONFIG_VALUE_2
)
```

Reply:
I think we need to stick to java8 here